samza-rest/src/main/java/org/apache/samza/monitor/SamzaMonitorService.java

```diff
@@ -100,4 +110,19 @@ public void run() {
       }
     };
   }
+
+  /**
+   * Creates a ScheduledThreadPoolExecutor with core pool size 1
+   * @return ScheduledExecutorService
+   */
+  private ScheduledExecutorService createScheduler() {
```

Comment:
Just curious.

Initially we had a single threaded scheduler to ensure when two monitors in samza-admin are trying to update/modify some shared state(either on disk or in some external store), then their executions are serialized. How are we planning to achieve that goal when we allow multiple monitors to be running at the same time in their own threads.

Reply:
We've already run into performance limits of what single threaded execution for all monitors can do with some monitors within LinkedIn, so while the current implementation avoided premature optimization, it's not sustainable. A better way to achieve the exclusivity goal will be to write the monitors with such side-effects in a "thread-safe" way. @shanthoosh are you aware of any existing monitors could be impacted by this change? If so, we can update them.

Shekhar, let's also call this out as another potentially backwards incompatible change in the "API  changes" section.

Reply:
>> are you aware of any existing monitors could be impacted by this change?
Let's please double-check if the parallel execution of monitors(StateStoreGCMonitor, StateStoreDeltaMonitor) internally could potentially cause any correctness issues. 

Other than that, updating the API spec of Monitor abstraction(& calling out backwards compatibility) should be sufficient.

Reply:
Ack, thanks for the heads up!